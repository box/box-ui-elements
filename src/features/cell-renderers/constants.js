// @flow

const EMPTY_VALUE = 'â€”';
const BYTES_IN_MB = 1048576;
const ANONYMOUS_USER_ID = '2';

export { EMPTY_VALUE, BYTES_IN_MB, ANONYMOUS_USER_ID };
