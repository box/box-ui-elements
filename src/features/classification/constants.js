// @flow
const DEFAULT_MAX_APP_COUNT = 3;

const SECURITY_CONTROLS_FORMAT: {
    FULL: 'full',
    SHORT: 'short',
    SHORT_WITH_BTN: 'shortWithBtn',
} = {
    FULL: 'full',
    SHORT: 'short',
    SHORT_WITH_BTN: 'shortWithBtn',
};

const ACCESS_POLICY_RESTRICTION: {
    APP: 'app',
    DOWNLOAD: 'download',
    EXTERNAL_COLLAB: 'externalCollab',
    FTP: 'ftp',
    SHARED_LINK: 'sharedLink',
    WATERMARK: 'watermark',
} = {
    SHARED_LINK: 'sharedLink',
    DOWNLOAD: 'download',
    EXTERNAL_COLLAB: 'externalCollab',
    APP: 'app',
    FTP: 'ftp',
    WATERMARK: 'watermark',
};

const DOWNLOAD_CONTROL: {
    DESKTOP: 'desktop',
    FTP: 'ftp',
    MOBILE: 'mobile',
    WEB: 'web',
} = {
    DESKTOP: 'desktop',
    FTP: 'ftp',
    MOBILE: 'mobile',
    WEB: 'web',
};

const LIST_ACCESS_LEVEL: {
    BLACKLIST: 'blacklist',
    BLOCK: 'block',
    WHITELIST: 'whitelist',
} = {
    BLACKLIST: 'blacklist',
    BLOCK: 'block',
    WHITELIST: 'whitelist',
};

const MANAGED_USERS_ACCESS_LEVEL: {
    OWNERS_AND_COOWNERS: 'ownersCoOwners',
    OWNERS_COOWNERS_AND_EDITORS: 'ownersCoOwnersEditors',
} = {
    OWNERS_AND_COOWNERS: 'ownersCoOwners',
    OWNERS_COOWNERS_AND_EDITORS: 'ownersCoOwnersEditors',
};

const SHARED_LINK_ACCESS_LEVEL: {
    COLLAB_AND_COMPANY_ONLY: 'companyAndCollabOnly',
    COLLAB_ONLY: 'collabOnly',
    PUBLIC: 'public',
} = {
    COLLAB_AND_COMPANY_ONLY: 'companyAndCollabOnly',
    COLLAB_ONLY: 'collabOnly',
    PUBLIC: 'public',
};

const APP_RESTRICTION_MESSAGE_TYPE = {
    DEFAULT: 'default',
    WITH_APP_LIST: 'withAppList',
    WITH_OVERFLOWN_APP_LIST: 'withOverflownAppList',
};

const CLASSIFICATION_COLOR_ID_0 = 0;
const CLASSIFICATION_COLOR_ID_1 = 1;
const CLASSIFICATION_COLOR_ID_2 = 2;
const CLASSIFICATION_COLOR_ID_3 = 3;
const CLASSIFICATION_COLOR_ID_4 = 4;
const CLASSIFICATION_COLOR_ID_5 = 5;
const CLASSIFICATION_COLOR_ID_6 = 6;
const CLASSIFICATION_COLOR_ID_7 = 7;

const DEFAULT_CLASSIFICATION_COLOR_ID = CLASSIFICATION_COLOR_ID_0;

export {
    ACCESS_POLICY_RESTRICTION,
    APP_RESTRICTION_MESSAGE_TYPE,
    CLASSIFICATION_COLOR_ID_0,
    CLASSIFICATION_COLOR_ID_1,
    CLASSIFICATION_COLOR_ID_2,
    CLASSIFICATION_COLOR_ID_3,
    CLASSIFICATION_COLOR_ID_4,
    CLASSIFICATION_COLOR_ID_5,
    CLASSIFICATION_COLOR_ID_6,
    CLASSIFICATION_COLOR_ID_7,
    DEFAULT_CLASSIFICATION_COLOR_ID,
    DEFAULT_MAX_APP_COUNT,
    DOWNLOAD_CONTROL,
    LIST_ACCESS_LEVEL,
    MANAGED_USERS_ACCESS_LEVEL,
    SECURITY_CONTROLS_FORMAT,
    SHARED_LINK_ACCESS_LEVEL,
};
