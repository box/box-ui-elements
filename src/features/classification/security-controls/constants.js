// @flow
const DEFAULT_MAX_APP_COUNT = 3;

const SECURITY_CONTROLS_FORMAT: {
    FULL: 'full',
    SHORT: 'short',
    SHORT_WITH_TOOLTIP: 'shortWithTooltip',
} = {
    FULL: 'full',
    SHORT: 'short',
    SHORT_WITH_TOOLTIP: 'shortWithTooltip',
};

const ACCESS_POLICY_RESTRICTION: {
    APP: 'app',
    DOWNLOAD: 'download',
    EXTERNAL_COLLAB: 'externalCollab',
    FTP: 'ftp',
    SHARED_LINK: 'sharedLink',
} = {
    SHARED_LINK: 'sharedLink',
    DOWNLOAD: 'download',
    EXTERNAL_COLLAB: 'externalCollab',
    APP: 'app',
    FTP: 'ftp',
};

const DOWNLOAD_CONTROL: {
    DESKTOP: 'desktop',
    FTP: 'ftp',
    MOBILE: 'mobile',
    WEB: 'web',
} = {
    DESKTOP: 'desktop',
    FTP: 'ftp',
    MOBILE: 'mobile',
    WEB: 'web',
};

const LIST_ACCESS_LEVEL: {
    BLACKLIST: 'blacklist',
    BLOCK: 'block',
    WHITELIST: 'whitelist',
} = {
    BLACKLIST: 'blacklist',
    BLOCK: 'block',
    WHITELIST: 'whitelist',
};

const MANAGED_USERS_ACCESS_LEVEL: {
    OWNERS_AND_COOWNERS: 'ownersCoOwners',
    OWNERS_COOWNERS_AND_EDITORS: 'ownersCoOwnersEditors',
} = {
    OWNERS_AND_COOWNERS: 'ownersCoOwners',
    OWNERS_COOWNERS_AND_EDITORS: 'ownersCoOwnersEditors',
};

const SHARED_LINK_ACCESS_LEVEL: {
    COLLAB_AND_COMPANY_ONLY: 'companyAndCollabOnly',
    COLLAB_ONLY: 'collabOnly',
    PUBLIC: 'public',
} = {
    COLLAB_AND_COMPANY_ONLY: 'companyAndCollabOnly',
    COLLAB_ONLY: 'collabOnly',
    PUBLIC: 'public',
};

export {
    ACCESS_POLICY_RESTRICTION,
    DEFAULT_MAX_APP_COUNT,
    DOWNLOAD_CONTROL,
    LIST_ACCESS_LEVEL,
    MANAGED_USERS_ACCESS_LEVEL,
    SECURITY_CONTROLS_FORMAT,
    SHARED_LINK_ACCESS_LEVEL,
};
