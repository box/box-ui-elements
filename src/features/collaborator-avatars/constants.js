// @flow

const COLLAB_GROUP_TYPE = 'group';
const COLLAB_USER_TYPE = 'user';
const COLLAB_PENDING_TYPE = 'pending';

export { COLLAB_GROUP_TYPE, COLLAB_USER_TYPE, COLLAB_PENDING_TYPE };
