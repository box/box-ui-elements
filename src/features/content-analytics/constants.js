// @flow
export const DOWNLOADS: 'DOWNLOADS' = 'DOWNLOADS';

export const PREVIEWS: 'PREVIEWS' = 'PREVIEWS';

export const USERS: 'USERS' = 'USERS';

export const METRIC = Object.freeze({
    DOWNLOADS,
    PREVIEWS,
    USERS,
});

const WEEK: 'week' = 'week';

const MONTH: 'month' = 'month';

const THREEMONTHS: 'threemonths' = 'threemonths';

const YEAR: 'year' = 'year';

export const PERIOD = Object.freeze({
    WEEK,
    MONTH,
    THREEMONTHS,
    YEAR,
});
