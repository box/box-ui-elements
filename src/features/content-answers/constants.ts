export const MOCK_LONG_PROMPT =
    'What is the significance of the character limit in the text area, and why is it important to ensure that users are notified when they exceed it? Additionally, what steps can be taken to prevent users from exceeding the character limit, and how can errors related to the character limit be handled in a user-friendly manner?';

export const TEXT_AREA = {
    MAX_LENGTH: 300,
};
