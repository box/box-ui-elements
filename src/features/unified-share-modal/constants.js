// @flow

// Shared link access level constants
const ANYONE_WITH_LINK: 'peopleWithTheLink' = 'peopleWithTheLink';
const ANYONE_IN_COMPANY: 'peopleInYourCompany' = 'peopleInYourCompany';
const PEOPLE_IN_ITEM: 'peopleInThisItem' = 'peopleInThisItem';

// Shared link permission level constants
const CAN_EDIT: 'canEdit' = 'canEdit';
const CAN_VIEW_DOWNLOAD: 'canViewDownload' = 'canViewDownload';
const CAN_VIEW_ONLY: 'canViewOnly' = 'canViewOnly';

// Invitee permission level constants
const EDITOR = 'Editor';
const CO_OWNER = 'Co-owner';
const PREVIEWER = 'Previewer';
const PREVIEWER_UPLOADER = 'Previewer Uploader';
const OWNER = 'Owner';
const VIEWER = 'Viewer';
const VIEWER_UPLOADER = 'Viewer Uploader';
const UPLOADER = 'Uploader';

const COLLAB_GROUP_TYPE = 'group';
const COLLAB_USER_TYPE = 'user';
const COLLAB_PENDING_TYPE = 'pending';

// Business Justfications for external collab restrictions
const JUSTIFICATION_CHECKPOINT_COLLAB: 'COLLAB' = 'COLLAB';
const JUSTIFICATION_CHECKPOINT_CREATE_SHARED_LINK: 'CREATE_SHARED_LINK' = 'CREATE_SHARED_LINK';
const JUSTIFICATION_CHECKPOINT_DOWNLOAD: 'DOWNLOAD' = 'DOWNLOAD';
const JUSTIFICATION_CHECKPOINT_EXTERNAL_COLLAB: 'EXTERNAL_COLLAB' = 'EXTERNAL_COLLAB';

// Default allowed access levels
const ALLOWED_ACCESS_LEVELS = {
    peopleInThisItem: true,
    peopleInYourCompany: true,
    peopleWithTheLink: true,
};

const DISABLED_REASON_ACCESS_POLICY: 'access_policy' = 'access_policy';
const DISABLED_REASON_MALICIOUS_CONTENT: 'malicious_content' = 'malicious_content';

const INVITEE_PERMISSIONS_FOLDER = [
    {
        default: false,
        text: CO_OWNER,
        value: CO_OWNER,
    },
    {
        default: true, // default in the WebApp
        text: EDITOR,
        value: EDITOR,
    },
    {
        default: false,
        text: PREVIEWER,
        value: PREVIEWER,
    },
    {
        default: false,
        text: PREVIEWER_UPLOADER,
        value: PREVIEWER_UPLOADER,
    },
    {
        default: false,
        text: UPLOADER,
        value: UPLOADER,
    },
    {
        default: false,
        text: VIEWER,
        value: VIEWER,
    },
    {
        default: false,
        text: VIEWER_UPLOADER,
        value: VIEWER_UPLOADER,
    },
];

const INVITEE_PERMISSIONS_FILE = [
    {
        default: true, // default in the WebApp
        text: EDITOR,
        value: EDITOR,
    },
    {
        default: false,
        text: VIEWER,
        value: VIEWER,
    },
];

export {
    ALLOWED_ACCESS_LEVELS,
    ANYONE_IN_COMPANY,
    ANYONE_WITH_LINK,
    CAN_EDIT,
    CAN_VIEW_DOWNLOAD,
    CAN_VIEW_ONLY,
    COLLAB_GROUP_TYPE,
    COLLAB_PENDING_TYPE,
    COLLAB_USER_TYPE,
    CO_OWNER,
    DISABLED_REASON_ACCESS_POLICY,
    DISABLED_REASON_MALICIOUS_CONTENT,
    EDITOR,
    INVITEE_PERMISSIONS_FOLDER,
    INVITEE_PERMISSIONS_FILE,
    JUSTIFICATION_CHECKPOINT_COLLAB,
    JUSTIFICATION_CHECKPOINT_CREATE_SHARED_LINK,
    JUSTIFICATION_CHECKPOINT_DOWNLOAD,
    JUSTIFICATION_CHECKPOINT_EXTERNAL_COLLAB,
    OWNER,
    PEOPLE_IN_ITEM,
    PREVIEWER,
    PREVIEWER_UPLOADER,
    UPLOADER,
    VIEWER,
    VIEWER_UPLOADER,
};
