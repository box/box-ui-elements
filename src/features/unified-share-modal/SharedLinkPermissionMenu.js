// @flow

import React, { Component } from 'react';
import classNames from 'classnames';
import { FormattedMessage } from 'react-intl';

import DropdownMenu, { MenuToggle } from '../../components/dropdown-menu';
import LabelPill from '../../components/label-pill';
import PlainButton from '../../components/plain-button';
import { Menu, SelectMenuItem } from '../../components/menu';
import type { TargetingApi } from '../targeting/types';

import type { permissionLevelType } from './flowTypes';
import { CAN_EDIT, CAN_VIEW_DOWNLOAD, CAN_VIEW_ONLY } from './constants';
import messages from './messages';

import './SharedLinkPermissionMenu.scss';

type Props = {
    allowedPermissionLevels: Array<permissionLevelType>,
    canChangePermissionLevel: boolean,
    changePermissionLevel: (
        newPermissionLevel: permissionLevelType,
    ) => Promise<{ permissionLevel: permissionLevelType }>,
    isSharedLinkEditTooltipShown: boolean,
    permissionLevel?: permissionLevelType,
    sharedLinkEditTagTargetingApi?: TargetingApi,
    sharedLinkEditTooltipTargetingApi?: TargetingApi,
    submitting: boolean,
    trackingProps: {
        onChangeSharedLinkPermissionLevel?: Function,
        sharedLinkPermissionsMenuButtonProps?: Object,
    },
};

class SharedLinkPermissionMenu extends Component<Props> {
    static defaultProps = {
        trackingProps: {},
    };

    onChangePermissionLevel = (newPermissionLevel: permissionLevelType) => {
        const { changePermissionLevel, permissionLevel, trackingProps } = this.props;
        const { onChangeSharedLinkPermissionLevel } = trackingProps;

        if (permissionLevel !== newPermissionLevel) {
            changePermissionLevel(newPermissionLevel);

            if (onChangeSharedLinkPermissionLevel) {
                onChangeSharedLinkPermissionLevel(newPermissionLevel);
            }
        }
    };

    render() {
        const {
            allowedPermissionLevels,
            isSharedLinkEditTooltipShown,
            permissionLevel,
            sharedLinkEditTagTargetingApi,
            sharedLinkEditTooltipTargetingApi,
            submitting,
            trackingProps,
        } = this.props;
        const { sharedLinkPermissionsMenuButtonProps } = trackingProps;
        const canShowTag = sharedLinkEditTagTargetingApi ? sharedLinkEditTagTargetingApi.canShow : false;
        const canShowTooltip = sharedLinkEditTooltipTargetingApi ? sharedLinkEditTooltipTargetingApi.canShow : false;

        if (!permissionLevel) {
            return null;
        }

        const permissionLevels = {
            [CAN_EDIT]: {
                label: <FormattedMessage {...messages.sharedLinkPermissionsEdit} />,
            },
            [CAN_VIEW_DOWNLOAD]: {
                label: <FormattedMessage {...messages.sharedLinkPermissionsViewDownload} />,
            },
            [CAN_VIEW_ONLY]: {
                label: <FormattedMessage {...messages.sharedLinkPermissionsViewOnly} />,
            },
        };

        return (
            <DropdownMenu
                constrainToWindow
                onMenuClose={() => {
                    if (allowedPermissionLevels.includes(CAN_EDIT) && canShowTag && sharedLinkEditTagTargetingApi) {
                        sharedLinkEditTagTargetingApi.onComplete();
                    }
                }}
                onMenuOpen={() => {
                    if (allowedPermissionLevels.includes(CAN_EDIT) && canShowTag && sharedLinkEditTagTargetingApi) {
                        sharedLinkEditTagTargetingApi.onShow();
                    }

                    // complete tooltip FTUX on opening of dropdown menu
                    if (isSharedLinkEditTooltipShown && canShowTooltip && sharedLinkEditTooltipTargetingApi) {
                        sharedLinkEditTooltipTargetingApi.onComplete();
                    }
                }}
            >
                <PlainButton
                    className={classNames('lnk', {
                        'is-disabled': submitting,
                        'bdl-is-disabled': submitting,
                    })}
                    disabled={submitting}
                    {...sharedLinkPermissionsMenuButtonProps}
                >
                    <MenuToggle>{permissionLevels[permissionLevel].label}</MenuToggle>
                </PlainButton>
                <Menu className="ums-share-permissions-menu">
                    {allowedPermissionLevels.map(level => (
                        <SelectMenuItem
                            key={level}
                            isSelected={level === permissionLevel}
                            onClick={() => this.onChangePermissionLevel(level)}
                        >
                            <div className="ums-share-permissions-menu-item">
                                <span>{permissionLevels[level].label}</span>
                                {level === CAN_EDIT && canShowTag && (
                                    <LabelPill.Pill className="ftux-editable-shared-link" type="ftux">
                                        <LabelPill.Text>
                                            <FormattedMessage {...messages.ftuxSharedLinkPermissionsEditTag} />
                                        </LabelPill.Text>
                                    </LabelPill.Pill>
                                )}
                            </div>
                        </SelectMenuItem>
                    ))}
                </Menu>
            </DropdownMenu>
        );
    }
}

export default SharedLinkPermissionMenu;
