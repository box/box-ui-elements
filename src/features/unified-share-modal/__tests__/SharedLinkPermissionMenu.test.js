import React from 'react';

import { CAN_EDIT, CAN_VIEW_DOWNLOAD, CAN_VIEW_ONLY } from '../constants';
import SharedLinkPermissionMenu from '../SharedLinkPermissionMenu';

describe('features/unified-share-modal/SharedLinkPermissionMenu', () => {
    const allowedPermissionLevels = [CAN_EDIT, CAN_VIEW_DOWNLOAD, CAN_VIEW_ONLY];
    const defaultSharedLinkEditTagTargetingApi = {
        canShow: false,
    };

    describe('render()', () => {
        [
            {
                submitting: true,
                permissionLevel: CAN_EDIT,
            },
            {
                submitting: false,
                permissionLevel: CAN_EDIT,
            },
            {
                submitting: true,
                permissionLevel: CAN_VIEW_DOWNLOAD,
            },
            {
                submitting: false,
                permissionLevel: CAN_VIEW_DOWNLOAD,
            },
            {
                submitting: true,
                permissionLevel: CAN_VIEW_ONLY,
            },
            {
                submitting: false,
                permissionLevel: CAN_VIEW_ONLY,
            },
        ].forEach(({ submitting, permissionLevel }) => {
            test('it should render correct menu', () => {
                const sharedLinkPermissionMenu = shallow(
                    <SharedLinkPermissionMenu
                        allowedPermissionLevels={allowedPermissionLevels}
                        canChangePermissionLevel
                        changePermissionLevel={() => {}}
                        permissionLevel={permissionLevel}
                        sharedLinkEditTagTargetingApi={defaultSharedLinkEditTagTargetingApi}
                        submitting={submitting}
                    />,
                );

                expect(sharedLinkPermissionMenu).toMatchSnapshot();
            });
        });

        test('should not render if permission level prop is not set', () => {
            const emptySharedLinkPermissionMenu = shallow(
                <SharedLinkPermissionMenu
                    allowedPermissionLevels={allowedPermissionLevels}
                    canChangePermissionLevel
                    changePermissionLevel={() => {}}
                    permissionLevel=""
                    sharedLinkEditTagTargetingApi={defaultSharedLinkEditTagTargetingApi}
                    submitting={false}
                />,
            );

            expect(emptySharedLinkPermissionMenu).toMatchSnapshot();
        });

        test.each`
            canShow  | length | should
            ${true}  | ${1}   | ${'should render LabelPillText if canShow is true'}
            ${false} | ${0}   | ${'should not render LabelPillText if canShow is false'}
        `('$should ', ({ canShow, length }) => {
            const wrapper = shallow(
                <SharedLinkPermissionMenu
                    allowedPermissionLevels={allowedPermissionLevels}
                    changePermissionLevel={() => {}}
                    permissionLevel={CAN_EDIT}
                    sharedLinkEditTagTargetingApi={{ canShow }}
                    submitting={false}
                />,
            );

            expect(wrapper.find('LabelPillText')).toHaveLength(length);
        });
    });

    describe('onChangePermissionLevel()', () => {
        test('should call tracking function and handler on menu change if it is set', () => {
            const changeMenuMock = jest.fn();
            const permissionLevelSpy = jest.fn();
            const sharedLinkPermissionMenu = shallow(
                <SharedLinkPermissionMenu
                    allowedPermissionLevels={allowedPermissionLevels}
                    canChangePermissionLevel={false}
                    changePermissionLevel={permissionLevelSpy}
                    permissionLevel={CAN_VIEW_DOWNLOAD}
                    sharedLinkEditTagTargetingApi={defaultSharedLinkEditTagTargetingApi}
                    submitting={false}
                    trackingProps={{
                        onChangeSharedLinkPermissionLevel: changeMenuMock,
                    }}
                />,
            );

            sharedLinkPermissionMenu.instance().onChangePermissionLevel(CAN_VIEW_ONLY);

            expect(changeMenuMock).toBeCalled();
            expect(permissionLevelSpy).toBeCalled();
        });

        test('should not call tracking function or handler on menu change if it is set (when the permissionLevel has not changed)', () => {
            const changeMenuMock = jest.fn();
            const permissionLevelSpy = jest.fn();
            const sharedLinkPermissionMenu = shallow(
                <SharedLinkPermissionMenu
                    allowedPermissionLevels={allowedPermissionLevels}
                    canChangePermissionLevel={false}
                    changePermissionLevel={permissionLevelSpy}
                    permissionLevel={CAN_VIEW_ONLY}
                    sharedLinkEditTagTargetingApi={defaultSharedLinkEditTagTargetingApi}
                    submitting={false}
                    trackingProps={{
                        onChangeSharedLinkPermissionLevel: changeMenuMock,
                    }}
                />,
            );

            sharedLinkPermissionMenu.instance().onChangePermissionLevel(CAN_VIEW_ONLY);

            expect(changeMenuMock).not.toBeCalled();
            expect(permissionLevelSpy).not.toBeCalled();
        });
    });
});
