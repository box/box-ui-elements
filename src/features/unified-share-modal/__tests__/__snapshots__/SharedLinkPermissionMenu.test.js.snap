// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features/unified-share-modal/SharedLinkPermissionMenu render() it should render correct menu 1`] = `
<DropdownMenu
  constrainToScrollParent={false}
  constrainToWindow={true}
  isRightAligned={false}
  onMenuClose={[Function]}
  onMenuOpen={[Function]}
>
  <PlainButton
    className="lnk is-disabled bdl-is-disabled"
    disabled={true}
  >
    <MenuToggle>
      <FormattedMessage
        defaultMessage="Can edit"
        id="boxui.unifiedShare.sharedLinkPermissionsEdit"
      />
    </MenuToggle>
  </PlainButton>
  <Menu
    className="ums-share-permissions-menu"
    isHidden={false}
    isSubmenu={false}
  >
    <SelectMenuItem
      isSelected={true}
      key="canEdit"
      onClick={[Function]}
    >
      <div
        className="ums-share-permissions-menu-item"
      >
        <span>
          <FormattedMessage
            defaultMessage="Can edit"
            id="boxui.unifiedShare.sharedLinkPermissionsEdit"
          />
        </span>
      </div>
    </SelectMenuItem>
    <SelectMenuItem
      isSelected={false}
      key="canViewDownload"
      onClick={[Function]}
    >
      <div
        className="ums-share-permissions-menu-item"
      >
        <span>
          <FormattedMessage
            defaultMessage="Can view and download"
            id="boxui.unifiedShare.sharedLinkPermissionsViewDownload"
          />
        </span>
      </div>
    </SelectMenuItem>
    <SelectMenuItem
      isSelected={false}
      key="canViewOnly"
      onClick={[Function]}
    >
      <div
        className="ums-share-permissions-menu-item"
      >
        <span>
          <FormattedMessage
            defaultMessage="Can view only"
            id="boxui.unifiedShare.sharedLinkPermissionsViewOnly"
          />
        </span>
      </div>
    </SelectMenuItem>
  </Menu>
</DropdownMenu>
`;

exports[`features/unified-share-modal/SharedLinkPermissionMenu render() it should render correct menu 2`] = `
<DropdownMenu
  constrainToScrollParent={false}
  constrainToWindow={true}
  isRightAligned={false}
  onMenuClose={[Function]}
  onMenuOpen={[Function]}
>
  <PlainButton
    className="lnk"
    disabled={false}
  >
    <MenuToggle>
      <FormattedMessage
        defaultMessage="Can edit"
        id="boxui.unifiedShare.sharedLinkPermissionsEdit"
      />
    </MenuToggle>
  </PlainButton>
  <Menu
    className="ums-share-permissions-menu"
    isHidden={false}
    isSubmenu={false}
  >
    <SelectMenuItem
      isSelected={true}
      key="canEdit"
      onClick={[Function]}
    >
      <div
        className="ums-share-permissions-menu-item"
      >
        <span>
          <FormattedMessage
            defaultMessage="Can edit"
            id="boxui.unifiedShare.sharedLinkPermissionsEdit"
          />
        </span>
      </div>
    </SelectMenuItem>
    <SelectMenuItem
      isSelected={false}
      key="canViewDownload"
      onClick={[Function]}
    >
      <div
        className="ums-share-permissions-menu-item"
      >
        <span>
          <FormattedMessage
            defaultMessage="Can view and download"
            id="boxui.unifiedShare.sharedLinkPermissionsViewDownload"
          />
        </span>
      </div>
    </SelectMenuItem>
    <SelectMenuItem
      isSelected={false}
      key="canViewOnly"
      onClick={[Function]}
    >
      <div
        className="ums-share-permissions-menu-item"
      >
        <span>
          <FormattedMessage
            defaultMessage="Can view only"
            id="boxui.unifiedShare.sharedLinkPermissionsViewOnly"
          />
        </span>
      </div>
    </SelectMenuItem>
  </Menu>
</DropdownMenu>
`;

exports[`features/unified-share-modal/SharedLinkPermissionMenu render() it should render correct menu 3`] = `
<DropdownMenu
  constrainToScrollParent={false}
  constrainToWindow={true}
  isRightAligned={false}
  onMenuClose={[Function]}
  onMenuOpen={[Function]}
>
  <PlainButton
    className="lnk is-disabled bdl-is-disabled"
    disabled={true}
  >
    <MenuToggle>
      <FormattedMessage
        defaultMessage="Can view and download"
        id="boxui.unifiedShare.sharedLinkPermissionsViewDownload"
      />
    </MenuToggle>
  </PlainButton>
  <Menu
    className="ums-share-permissions-menu"
    isHidden={false}
    isSubmenu={false}
  >
    <SelectMenuItem
      isSelected={false}
      key="canEdit"
      onClick={[Function]}
    >
      <div
        className="ums-share-permissions-menu-item"
      >
        <span>
          <FormattedMessage
            defaultMessage="Can edit"
            id="boxui.unifiedShare.sharedLinkPermissionsEdit"
          />
        </span>
      </div>
    </SelectMenuItem>
    <SelectMenuItem
      isSelected={true}
      key="canViewDownload"
      onClick={[Function]}
    >
      <div
        className="ums-share-permissions-menu-item"
      >
        <span>
          <FormattedMessage
            defaultMessage="Can view and download"
            id="boxui.unifiedShare.sharedLinkPermissionsViewDownload"
          />
        </span>
      </div>
    </SelectMenuItem>
    <SelectMenuItem
      isSelected={false}
      key="canViewOnly"
      onClick={[Function]}
    >
      <div
        className="ums-share-permissions-menu-item"
      >
        <span>
          <FormattedMessage
            defaultMessage="Can view only"
            id="boxui.unifiedShare.sharedLinkPermissionsViewOnly"
          />
        </span>
      </div>
    </SelectMenuItem>
  </Menu>
</DropdownMenu>
`;

exports[`features/unified-share-modal/SharedLinkPermissionMenu render() it should render correct menu 4`] = `
<DropdownMenu
  constrainToScrollParent={false}
  constrainToWindow={true}
  isRightAligned={false}
  onMenuClose={[Function]}
  onMenuOpen={[Function]}
>
  <PlainButton
    className="lnk"
    disabled={false}
  >
    <MenuToggle>
      <FormattedMessage
        defaultMessage="Can view and download"
        id="boxui.unifiedShare.sharedLinkPermissionsViewDownload"
      />
    </MenuToggle>
  </PlainButton>
  <Menu
    className="ums-share-permissions-menu"
    isHidden={false}
    isSubmenu={false}
  >
    <SelectMenuItem
      isSelected={false}
      key="canEdit"
      onClick={[Function]}
    >
      <div
        className="ums-share-permissions-menu-item"
      >
        <span>
          <FormattedMessage
            defaultMessage="Can edit"
            id="boxui.unifiedShare.sharedLinkPermissionsEdit"
          />
        </span>
      </div>
    </SelectMenuItem>
    <SelectMenuItem
      isSelected={true}
      key="canViewDownload"
      onClick={[Function]}
    >
      <div
        className="ums-share-permissions-menu-item"
      >
        <span>
          <FormattedMessage
            defaultMessage="Can view and download"
            id="boxui.unifiedShare.sharedLinkPermissionsViewDownload"
          />
        </span>
      </div>
    </SelectMenuItem>
    <SelectMenuItem
      isSelected={false}
      key="canViewOnly"
      onClick={[Function]}
    >
      <div
        className="ums-share-permissions-menu-item"
      >
        <span>
          <FormattedMessage
            defaultMessage="Can view only"
            id="boxui.unifiedShare.sharedLinkPermissionsViewOnly"
          />
        </span>
      </div>
    </SelectMenuItem>
  </Menu>
</DropdownMenu>
`;

exports[`features/unified-share-modal/SharedLinkPermissionMenu render() it should render correct menu 5`] = `
<DropdownMenu
  constrainToScrollParent={false}
  constrainToWindow={true}
  isRightAligned={false}
  onMenuClose={[Function]}
  onMenuOpen={[Function]}
>
  <PlainButton
    className="lnk is-disabled bdl-is-disabled"
    disabled={true}
  >
    <MenuToggle>
      <FormattedMessage
        defaultMessage="Can view only"
        id="boxui.unifiedShare.sharedLinkPermissionsViewOnly"
      />
    </MenuToggle>
  </PlainButton>
  <Menu
    className="ums-share-permissions-menu"
    isHidden={false}
    isSubmenu={false}
  >
    <SelectMenuItem
      isSelected={false}
      key="canEdit"
      onClick={[Function]}
    >
      <div
        className="ums-share-permissions-menu-item"
      >
        <span>
          <FormattedMessage
            defaultMessage="Can edit"
            id="boxui.unifiedShare.sharedLinkPermissionsEdit"
          />
        </span>
      </div>
    </SelectMenuItem>
    <SelectMenuItem
      isSelected={false}
      key="canViewDownload"
      onClick={[Function]}
    >
      <div
        className="ums-share-permissions-menu-item"
      >
        <span>
          <FormattedMessage
            defaultMessage="Can view and download"
            id="boxui.unifiedShare.sharedLinkPermissionsViewDownload"
          />
        </span>
      </div>
    </SelectMenuItem>
    <SelectMenuItem
      isSelected={true}
      key="canViewOnly"
      onClick={[Function]}
    >
      <div
        className="ums-share-permissions-menu-item"
      >
        <span>
          <FormattedMessage
            defaultMessage="Can view only"
            id="boxui.unifiedShare.sharedLinkPermissionsViewOnly"
          />
        </span>
      </div>
    </SelectMenuItem>
  </Menu>
</DropdownMenu>
`;

exports[`features/unified-share-modal/SharedLinkPermissionMenu render() it should render correct menu 6`] = `
<DropdownMenu
  constrainToScrollParent={false}
  constrainToWindow={true}
  isRightAligned={false}
  onMenuClose={[Function]}
  onMenuOpen={[Function]}
>
  <PlainButton
    className="lnk"
    disabled={false}
  >
    <MenuToggle>
      <FormattedMessage
        defaultMessage="Can view only"
        id="boxui.unifiedShare.sharedLinkPermissionsViewOnly"
      />
    </MenuToggle>
  </PlainButton>
  <Menu
    className="ums-share-permissions-menu"
    isHidden={false}
    isSubmenu={false}
  >
    <SelectMenuItem
      isSelected={false}
      key="canEdit"
      onClick={[Function]}
    >
      <div
        className="ums-share-permissions-menu-item"
      >
        <span>
          <FormattedMessage
            defaultMessage="Can edit"
            id="boxui.unifiedShare.sharedLinkPermissionsEdit"
          />
        </span>
      </div>
    </SelectMenuItem>
    <SelectMenuItem
      isSelected={false}
      key="canViewDownload"
      onClick={[Function]}
    >
      <div
        className="ums-share-permissions-menu-item"
      >
        <span>
          <FormattedMessage
            defaultMessage="Can view and download"
            id="boxui.unifiedShare.sharedLinkPermissionsViewDownload"
          />
        </span>
      </div>
    </SelectMenuItem>
    <SelectMenuItem
      isSelected={true}
      key="canViewOnly"
      onClick={[Function]}
    >
      <div
        className="ums-share-permissions-menu-item"
      >
        <span>
          <FormattedMessage
            defaultMessage="Can view only"
            id="boxui.unifiedShare.sharedLinkPermissionsViewOnly"
          />
        </span>
      </div>
    </SelectMenuItem>
  </Menu>
</DropdownMenu>
`;

exports[`features/unified-share-modal/SharedLinkPermissionMenu render() should not render if permission level prop is not set 1`] = `""`;
